backend-api/src/functions/common/session/sessionRegistry.ts

```diff
@@ -0,0 +1,30 @@
+import { IUpdateSessionOperation, Session, SessionState } from "./Session";
+import { Result } from "../../utils/result";
+import { CreateSessionAttributes } from "../../services/session/sessionService";
+import {
+  DatabaseRecord,
+  ReadSessionError,
+} from "../../adapters/dynamoDbAdapter";
+
+export interface SessionRegistry {
+  getSessionWithState(
+    sessionId: string,
+    sessionState: SessionState,
+  ): Promise<Result<Session, ReadSessionError>>;
+
+  // Ideally the return value of this would be more dynamo-agnostic, to keep it in this interface
```

Comment:
Yes - I agree, and I'd prefer this. This interface would ideally return the AuthSessionCreatedSession I think?

Reply:
Yep that's what I was thinking